bundles/org.openhab.binding.plugwiseha/src/main/java/org/openhab/binding/plugwiseha/internal/handler/PlugwiseHABaseHandler.java

```diff
@@ -0,0 +1,220 @@
+ *
+ * @param <E> entity - the Plugwise Home Automation entity class used by this
+ *            thing handler
+ * @param <C> config - the Plugwise Home Automation config class used by this
+ *            thing handler
+ */
+
+@NonNullByDefault
+public abstract class PlugwiseHABaseHandler<E, C extends PlugwiseHAThingConfig> extends BaseThingHandler {
+
+    protected static final String STATUS_DESCRIPTION_COMMUNICATION_ERROR = "Error communicating with the Plugwise Home Automation controller";
+
+    protected final Logger logger = LoggerFactory.getLogger(PlugwiseHABaseHandler.class);
+
+    private Class<?> clazz;
+
+    // Constructor
+
+    public PlugwiseHABaseHandler(Thing thing) {
+        super(thing);
+        clazz = (Class<?>) (((ParameterizedType) getClass().getGenericSuperclass()).getActualTypeArguments()[1]);
+    }
+
+    // Abstract methods
+
+    /**
+     * Initializes the Plugwise Entity that this class handles.
+     *
+     * @param config the thing configuration
+     * @param bridge the bridge that this thing is part of
+     */
+    protected abstract void initialize(C config, PlugwiseHABridgeHandler bridge);
+
+    /**
+     * Get the Plugwise Entity that this class handles.
+     *
+     * @param controller the controller for this ThingHandler
+     * @param forceRefresh indicated if the entity should be refreshed from the Plugwise API
+     */
+    protected abstract @Nullable E getEntity(PlugwiseHAController controller, Boolean forceRefresh)
+            throws PlugwiseHAException;
+
+    /**
+     * Handles a {@link RefreshType} command for a given channel.
+     *
+     * @param entity the Plugwise Entity
+     * @param channelUID the channel uid the command is for
+     */
+    protected abstract void refreshChannel(E entity, ChannelUID channelUID);
+
+    /**
+     * Handles a command for a given channel.
+     * 
+     * @param entity the Plugwise Entity
+     * @param channelUID the channel uid the command is for
+     * @param command the command
+     */
+    protected abstract void handleCommand(E entity, ChannelUID channelUID, Command command) throws PlugwiseHAException;
+
+    // Overrides
+
+    @Override
+    public void initialize() {
+        C config = getPlugwiseThingConfig();
+
+        if (checkConfig(config)) {
+            Bridge bridge = getBridge();
+            if (bridge == null || bridge.getHandler() == null
+                    || !(bridge.getHandler() instanceof PlugwiseHABridgeHandler)) {
+                updateStatus(OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "You must choose a Plugwise Home Automation bridge for this thing.");
+                return;
+            }
+
+            if (bridge.getStatus() == OFFLINE) {
+                updateStatus(OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                        "The Plugwise Home Automation bridge is currently offline.");
+                return;
+            }
```

Comment:
If you return here the `initialize` method below will never be called when the bridge goes online.